homeassistant/components/shelly/__init__.py

```diff
@@ -90,6 +91,7 @@
     Platform.LIGHT,
     Platform.SENSOR,
     Platform.SWITCH,
+    Platform.UPDATE,
```

Comment:
The firmware update sensor is based on polling status from device, this needs to be carefully handled.

Reply:
I guess this should be respected by disabling the update entities by default 🤔

Reply:
I am not sure, looking few lines above we added `Platform.BUTTON` only to `BLOCK_PLATFORMS` and did not add it to `BLOCK_SLEEPING_PLATFORMS`, so we intentionally did not create OTA buttons for sleeping devices.
Will try to check it later today.

Reply:
Even for polling entities we should limit updates to at an hour maybe ? There will be hardly any data. I wonder for sleeping devices we can do it once a day in response to when they receive data (so we know device is active)

Reply:
That's correct - we only support powered (_non sleeping_ ) devices for update buttons, therefore I added the update entities also only to the non-sleeping devices

Reply:
> Even for polling entities we should limit updates to at an hour maybe ? There will be hardly any data. I wonder for sleeping devices we can do it once a day in response to when they receive data (so we know device is active)

That would be good but we currently poll REST entities every 60 seconds since there is also RSSI and other sensors that we wanted to update faster.

Reply:
Can we make a separate polling interface for update entities ? Would be weird that an update entity triggers a 60s polling cycle.

Reply:
And if they are on their own cycle and once an hour, they wouldn't have to be disabled by default.

I would prefer update entities to not have to be disabled by default.